From 7471b01916dee4faa05c933a1c62f1206c3fa7b7 Mon Sep 17 00:00:00 2001
From: Nishanth Menon <nm@ti.com>
Date: Mon, 14 Feb 2011 12:44:27 +0530
Subject: [PATCH 41/45] omap3+: sr: free name pointer if fail

name is allocated with kasprintf, so if we fail, we need to free
it as well.

Signed-off-by: Nishanth Menon <nm@ti.com>
---
 arch/arm/mach-omap2/smartreflex.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/smartreflex.c b/arch/arm/mach-omap2/smartreflex.c
index 3102240..4a2119d 100644
--- a/arch/arm/mach-omap2/smartreflex.c
+++ b/arch/arm/mach-omap2/smartreflex.c
@@ -299,6 +299,7 @@ error:
 		"interrupt handler. Smartreflex will"
 		"not function as desired\n", __func__);
 	kfree(sr_info);
+	kfree(name);
 	return ret;
 }
 
-- 
1.7.1

